web/styles/rendered_markdown.css

```diff
@@ -235,10 +235,16 @@
         border: hsl(0deg 0% 50%) 1px solid;
         padding: 2px 8px 2px 10px;
         border-radius: 10px;
-        position: relative;
-        top: 1px;
-        display: block;
-        margin: 5px 0 15px;
+        /* Space any subsequent Markdown content the same
+           distance as adjacent paragraphs are spaced. */
+        margin: 0 0 10px;
+
+        &:last-child {
+            /* Otherwise, a spoiler block at the end of a message,
+               or as a message's only content, gets the same bottom
+               margin as other elements. */
+            margin-bottom: 5px;
+        }
```

Comment:
Doesn't this still apply to spoilers nested inside a blockquote or something? We still need the direct descendant selector, I think... and probably there's still going to be quirks like a single-line spoiler inside a blockquote having extra bottom margin.

Reply:
So this is going to be outside the scope of the PR, but it looks like when a spoiler block is quoted, somehow a break tag and a spare paragraph tag get added:

![Screenshot 2024-04-18 at 6 00 30 PM](https://github.com/zulip/zulip/assets/170719/faaa7c62-5a8b-49db-b0dd-b9a74ac10840)

So while that will prevent this `:last-child` selector from applying in that particular case (I agree for the need to use the direct descendant selector), this is a case where we need to examine how that markup is coming into being.

For reference, this is the original message markup from that screenshot:

`````txt
````quote
```spoiler How does this look?
Let's find out.
```
````
`````

Reply:
This is how it renders. The bold is lost on the spoiler header, too.

![Screenshot 2024-04-18 at 6 06 28 PM](https://github.com/zulip/zulip/assets/170719/df4ee715-ab99-42c0-99f3-d8df317179ba)

Reply:
This appears to be an issue with quoted standalone code blocks and KaTeX, too. I don't fully understand `fenced_code.py`, but it looks as though it's doing some kind of redundant processing that's inserting extra breaks and paragraphs around the HTML of those elements.

Reply:
Updated this PR to use a direct descendant selector for spoilers, and also opened [a CZO discussion](https://chat.zulip.org/#narrow/stream/3-backend/topic/extra.20markup.20on.20quoted.20fences/near/1784077) about the Markdown issues here.